hist/hist/src/TGraph.cxx

```diff
@@ -2455,36 +2456,28 @@ Double_t **TGraph::ShrinkAndCopy(Int_t size, Int_t oend)
 ///   graph->Sort(&CompareErrors, kFALSE);
 /// ~~~
 
-void TGraph::Sort(Bool_t (*greaterfunc)(const TGraph*, Int_t, Int_t) /*=TGraph::CompareX()*/,
-                  Bool_t ascending /*=kTRUE*/, Int_t low /* =0 */, Int_t high /* =-1111 */)
+void TGraph::Sort(Bool_t (*greaterfunc)(const TGraph *, Int_t, Int_t) /*=TGraph::CompareX()*/,
+                  Bool_t ascending /*=kTRUE*/, Int_t low /*=0*/, Int_t high /*=-1111*/)
 {
-
    // set the bit in case of an ascending =sort in X
-   if (greaterfunc == TGraph::CompareX && ascending  && low == 0 && high == -1111)
+   if (greaterfunc == TGraph::CompareX && ascending && low == 0 && high == -1111)
       SetBit(TGraph::kIsSortedX);
 
-   if (high == -1111) high = GetN() - 1;
-   //  Termination condition
-   if (high <= low) return;
+   if (high == -1111)
+      high = fNpoints - 1;
 
-   int left, right;
-   left = low; // low is the pivot element
-   right = high;
-   while (left < right) {
-      // move left while item < pivot
-      while (left <= high && greaterfunc(this, left, low) != ascending)
-         left++;
-      // move right while item > pivot
-      while (right > low && greaterfunc(this, right, low) == ascending)
-         right--;
-      if (left < right && left < high && right > low)
-         SwapPoints(left, right);
-   }
-   // right is final position for the pivot
-   if (right > low)
-      SwapPoints(low, right);
-   Sort(greaterfunc, ascending, low, right - 1);
-   Sort(greaterfunc, ascending, right + 1, high);
+   // Create a vector to store the indices of the graph data points.
+   // We use std::vector<Int_t> instead of std::vector<ULong64_t> to match the input type
+   // required by the comparison operator's signature provided as `greaterfunc`
```

Comment:
Could you also add a comment here about the reason why we needed to use `stable_sort`? Namely, the `std::sort` implementation of libc++ is not standard-compliant until LLVM 14 and this was probably the cause of errors on mac nodes of our CI. We can point to the related github issue https://github.com/llvm/llvm-project/issues/21211